### PR TITLE
Remove remaining instances of sprintf outside of external_codes

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

This removes remaining instances of sprintf.

This fixes https://github.com/QMCPACK/qmcpack/issues/4438.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Code style update (formatting, renaming)
- Refactoring (no functional changes, no api changes)
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

My laptop, Ubuntu 20.04

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed

### CODE REVIEW

**Comment by markdewing** [comment]
I think it would be better if updates to Catch were in a separate PR.  Especially since this updates the version of Catch, and is more involved than a simple patch.

**Comment by quantumsteve** [comment]
> I think it would be better if updates to Catch were in a separate PR. Especially since this updates the version of Catch, and is more involved than a simple patch.

I removed the second commit and will create another PR.

File: src/QMCWaveFunctions/BsplineFactory/HybridRepSetReader.h [hunk 1]
```diff
@@ -561,51 +558,6 @@ class HybridRepSetReader : public SplineSetReader<SA>
           }
         }
       }
-
-#ifdef PRINT_RADIAL
-      char fname[64];
```
Comment (markdewing) [inline src/QMCWaveFunctions/BsplineFactory/HybridRepSetReader.h:566]: Is @ye-luo okay with this debug code being deleted?

Comment (ye-luo) [reply src/QMCWaveFunctions/BsplineFactory/HybridRepSetReader.h:566]: Fine with the deletion. I can restore these lines if I really need them.

Comment (quantumsteve) [reply src/QMCWaveFunctions/BsplineFactory/HybridRepSetReader.h:566]: I tried defining PRINT_RADIAL and the debug code is broken (`center_idx` isn't defined)

**Comment by markdewing** [comment]
Test this please

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**